server/app/modules/DatabaseSeedModule.java

```diff
@@ -23,17 +21,12 @@ public static final class DatabaseSeedScheduler {
 
     @Inject
     public DatabaseSeedScheduler(
-        ActorSystem actorSystem,
-        ExecutionContext executionContext,
+        ApplicationEvolutions applicationEvolutions,
         Provider<DatabaseSeedTask> databaseSeedTaskProvider) {
-      actorSystem
-          .scheduler()
-          .scheduleOnce(
-              // schedule seed task for 5 sec from now. There is a race condition
-              // with Play evolutions. Evolutions must run before we seed database.
-              // It doesn't seem to be a way to run code after evolutions so just
-              // give them few sec to run.
-              Duration.ofSeconds(5), () -> databaseSeedTaskProvider.get().run(), executionContext);
+
+      if (applicationEvolutions.upToDate()) {
```

Comment:
The note section above has more details and links. According to Matthias over with Play modules get reset each time an evolution is applied so we don't need to do any polling. Play will handle it.

Reply:
I think this might only apply to dev. I was confused about the lack of mentions about production so I looked into the docs and found this bit.
```
When evolutions are activated, Play will check your database schema state before each request in DEV mode, or before starting the application in PROD mode. In DEV mode, if your database schema is not up to date, an error page will suggest that you synchronize your database schema by running the appropriate SQL script.
```
If I'm understanding this correctly, this race condition would not be an issue in production since prod mode waits for all the evolutions to be applied to start the app (as opposed to dev mode which starts the app... whenever? unless you have this `upToDate` check applied).

Reply:
Would you add a comment or a bit in our wiki that explains why we need the `upToDate` method and/or links to that part of the migration guide?

Reply:
@rockycodes I added a small comment with a link to the PR.